Feature/interface plagin initializer

bhaalgore opened the pull request:
Понял свою ошибку в ходе написания подробностей по поводу того, что я сделал. Нужно делать коммиты чаще, я это обязательно учту.

bhaalgore committed:
add base_interface and gui_interface classes, minor fix of CMakelist

Добавил класс, от которого будут наследоваться gui_interface и класс, реализующий интерфейс для логики. Также добавил класс gui_interface. Изменил формирование переменных HEADERS и SOURCERS, теперь они формируются с помощью выражения GLOB_RECURSE. Добавил пустой json файл.

unitedtimur commented on core/CMakeLists.txt:
```diff
@@ -1,7 +1,8 @@
-set(SOURCES ${CMAKE_CURRENT_LIST_DIR}/src/core.cpp)
+file(GLOB_RECURSE SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)
+file(GLOB_RECURSE HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/*.h)
+#set(HEADERS ${CMAKE_CURRENT_LIST_DIR}/include/core.h ${CMAKE_CURRENT_LIST_DIR}/include/base_interface.h ${CMAKE_CURRENT_LIST_DIR}/include/namespace.h ${CMAKE_CURRENT_LIST_DIR}/include/gui_interface.h)
```
rem

unitedtimur commented on core/include/base_interface.h:
```diff
@@ -0,0 +1,23 @@
+#include <QPointer>
+
+namespace base_interface_namespace {
+
```
rem

unitedtimur commented on core/include/base_interface.h:
```diff
@@ -0,0 +1,23 @@
+#include <QList>
+#include <QPointer>
+
+namespace base_interface_namespace {
```
Namespace должен быть общим, то есть `Core`

unitedtimur commented on core/include/gui_interface.h:
```diff
@@ -0,0 +1,22 @@
+
+#include "base_interface.h"
+
+namespace gui_interface_namespace {
```
Аналогично

unitedtimur commented on core/include/gui_interface.h:
```diff
@@ -0,0 +1,22 @@
+#ifndef GUI_INTERFACE_H
+#define GUI_INTERFACE_H
+
+
```
Лишние пробелы

unitedtimur commented on plugins/interface/include/metadata.json:
```diff
@@ -0,0 +1 @@
+{}
```
А зачем тут `.json` сейчас? Понятное дело он будет нужен, но сейчас нет сорцов. Следовательно на данный момент этот файл - мусор.

bhaalgore committed:
minor fixes

Убраны закоменченые строки кода в CMakeLists. У всех классов в core теперь один namespace. Убраны инклюды для классов QList и QPointer.

bhaalgore committed:
fix namespace naming

unitedtimur commented on core/include/gui_interface.h:
```diff
@@ -0,0 +1,20 @@
+#include "base_interface.h"
+
+namespace Сore {
+
```
rem empty space

unitedtimur commented on core/include/gui_interface.h:
```diff
@@ -0,0 +1,20 @@
+        virtual ~GuiInterface() = default;
+        virtual bool initialize(const QList<QPointer<QObject>> &dependencies);
+    };
+
```
rem empty space

unitedtimur commented on core/include/gui_interface.h:
```diff
@@ -0,0 +1,20 @@
+    };
+
+}
+Q_DECLARE_INTERFACE(Сore::GuiInterface, "com.core.GuiInterface")
```
add `\n` above

unitedtimur commented on core/include/base_interface.h:
```diff
@@ -0,0 +1,18 @@
+        virtual bool initialize(const QList<QPointer<QObject>> &dependencies) = 0;
+    };
+}
+Q_DECLARE_INTERFACE(Сore::BaseInterface, "com.сore.BaseInterface")
```
add `\n` above

bhaalgore committed:
delete json-file

unitedtimur commented on core/include/base_interface.h:
```diff
@@ -0,0 +1,18 @@
+
+    public:
+        using QObject::QObject;
+        virtual ~BaseInterface() = default;
```
add `\n` above

unitedtimur commented on core/include/gui_interface.h:
```diff
@@ -0,0 +1,20 @@
+
+    public:
+        virtual ~GuiInterface() = default;
+        virtual bool initialize(const QList<QPointer<QObject>> &dependencies);
```
Не надо эту функцию сюда. Удаляй. Она будет имплементироваться в плагинах только.

unitedtimur commented on core/include/gui_interface.h:
```diff
@@ -0,0 +1,20 @@
+#ifndef GUI_INTERFACE_H
+#define GUI_INTERFACE_H
+#include "base_interface.h"
```
add `\n` above

unitedtimur commented on core/include/base_interface.h:
```diff
@@ -0,0 +1,18 @@
+#include <QObject>
+
+namespace Сore {
+    class BaseInterface : public QObject
```
Надо написать комментарии ко всем классам и методу, что они значат. В стиле doxygen. Комментарии с большой буквы

```
/*!
* \brief
*/
```

bhaalgore committed:
fix empty spaces

bhaalgore committed:
fix spacing

bhaalgore committed:
Merge remote-tracking branch 'origin/master' into feature/interface-plagin-initializer

unitedtimur commented on core/CMakeLists.txt:
```diff
@@ -1,4 +1,6 @@
 project(core)
+file(GLOB_RECURSE SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)
+file(GLOB_RECURSE HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/*.h)
```
Лишнее.

unitedtimur commented on core/include/base_interface.h:
```diff
@@ -0,0 +1,17 @@
+#ifndef BASE_INTERFACE_H
+#define BASE_INTERFACE_H
+#include <QObject>
```
add `\n` above

unitedtimur commented on core/include/base_interface.h:
```diff
@@ -0,0 +1,17 @@
+#ifndef BASE_INTERFACE_H
+#define BASE_INTERFACE_H
+#include <QObject>
+namespace Сore {
```
add `\n` above

unitedtimur commented on core/include/gui_interface.h:
```diff
@@ -0,0 +1,17 @@
+#ifndef GUI_INTERFACE_H
+#define GUI_INTERFACE_H
+#include "base_interface.h"
+namespace Сore {
```
add `\n` above

unitedtimur commented on core/include/gui_interface.h:
```diff
@@ -0,0 +1,17 @@
+
+    public:
+        virtual ~GuiInterface() = default;
+        virtual bool initialize(const QList<QPointer<QObject>> &dependencies);
```
Зачем тебе метод в этом интерфейсе? Удали его.

bhaalgore committed:
fix spacing, fix double using functions

bhaalgore committed:
add brief explanation of classes and their methods
